Manual Numark Mixtrack Platinum FX

evoixmr opened the pull request:
Manual for https://github.com/mixxxdj/mixxx/pull/12872

Preview:
https://deploy-preview-623--mixxx-manual.netlify.app/hardware/controllers/numark_mixtrack_platinum_fx

ronso0 commented:
Just a quick note regrading the images:
it would be preferred to create rather small SVG overviews instead of adding ~1 MB PNG images to the repo. You can create black/white SVGs by tracing the origial raster image (as you did in an earlier commit I think).

evoixmr commented:
@ronso0 I updated all photos and scripts to use svg. I also deleted the png files.

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,866 @@
+.. Credit to PopHippy for creating the original PDF file.
+
+Numark Mixtrack Platinum FX
+
```
This blank line before the dash line indicating a header makes the page generator pick **Overview** as page title.
Please remove.
![image](https://github.com/mixxxdj/manual/assets/5934199/088cef39-c869-4d5c-b767-e5b4223066ca)

evoixmr replied:
Updated header.

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,866 @@
+
+Mixtrack Platinum FX is a 24-bit 4-deck digital DJ controller with a dedicated software effects section, large 6-inch capacitive-touch jog wheels with built in hi-res displays and 16-multifunctional performance pads.
+
+   Firmware
+
+   The mapping requires the following Soft/Firmware for all enabled
+   features to work for Mixxx 2.4
+
+-  `Firmware Update v1.10 MAC <https://cdn.inmusicbrands.com/Numark/MixtrackPlatinumFX/Mixtrack%20Platinum%20FX%20Firmware%201.10%20Mac.zip>`__
+-  `Firmware Update v1.10 WIN <https://cdn.inmusicbrands.com/Numark/MixtrackPlatinumFX/Mixtrack%20Platinum%20FX%20Firmware%201.10%20Win.zip>`__
+- Linux user need to install firmware using Windows or MAC.
```
```suggestion
Firmware
--------

The mapping requires the following Soft/Firmware for all enabled features to work for Mixxx 2.4

-  `Firmware Update v1.10 MAC <https://cdn.inmusicbrands.com/Numark/MixtrackPlatinumFX/Mixtrack%20Platinum%20FX%20Firmware%201.10%20Mac.zip>`__
-  `Firmware Update v1.10 WIN <https://cdn.inmusicbrands.com/Numark/MixtrackPlatinumFX/Mixtrack%20Platinum%20FX%20Firmware%201.10%20Win.zip>`__
- Linux user need to install firmware using Windows or MAC.
```

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,866 @@
+   :alt: Numark Mixtrack Playinum FX (schematic view)
+   :figclass: pretty-figures
+
+---------------
```
```suggestion
---------------
```
Please remove these lines. They prevent proper header rendering. Also some table rows are broken due to blank lines.
I suggest you check the preview and tweak until it looks as desired.
https://deploy-preview-623--mixxx-manual.netlify.app/hardware/controllers/numark_mixtrack_platinum_fx

evoixmr replied:
I removed the lines.

ronso0 replied:
The appearance is still broken, just check the preview.
FYI you may also build the html locally with `make html` and double-check before pushing.

evoixmr replied:
I just updated the appearance. Can you recheck and make suggestion on which areas need to be improve?

ronso0 replied:
Yes, it's loking good now. It were just the blank lines that enforced header formatting on each following line.

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,830 @@
+which sets the effective bpm to the one tapped using the 
+tempo adjust. Shift tap resets to 0 tempo change. 
+
+The alternative changes the actual file bpm. The problem 
```
This is not a problem, it's the effect of changing the file BPM.

Simply add a hint where users can change the BPM manually, the Track Properties dialog and the BPM column (if inline editing is enabled in Preferences -> Library).

evoixmr replied:
Deleted

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,830 @@
+accessible by setting a code variable MixtrackPlatinumFX.tapChangesTempo to 
+false, recommended only for advanced users.
+ 
+..
```
??

evoixmr replied:
Deleted

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,830 @@
+ 
+..
+
+For the default, the MIXXX common function takes taps and averages
+them. If you don't tap for 2 seconds the average resets and you start
+again. To prevent accidental double taps or misses if a tap is 40%
+shorter or 80% longer it will be ignored. I found while testing
+sometimes if I got the first two taps wrong the rest would be
+rejected by the filter, but it wasn't obvious this was happening. By
+default the button is dimly lit (like most others). When you tap the
+button if it accepts the tap it will go bright, if it rejects it from
+the filter it will show off. If this happens stop tapping wait 2
+seconds for the filter to clear and try again. Fixed this kicking in
+after 8 taps.
```
Does this apply to your custom BPM tap function only?
If no, I don't think this information belongs here. Rather add that hint to https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-bpm_tap (in a separate PR) and link to it with
```
:mixxx:coref:`bpm_tap<[ChannelN],bpm_tap>`
```

evoixmr replied:
Deleted

evoixmr commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,830 @@
+   | d.Fade Cuts Mode & Layers
+   | e.Sample Mode & Layers
+   | 6.Advanced User Section
+   | 7.Notes
```
Deleted

evoixmr commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,830 @@
+|                       |      mid-range        |                       |
+|                       |      frequencies for  |                       |
+|                       |      the individual   |                       |
+|                       |      channels.        |                       |
```
Updated appearance

evoixmr commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,830 @@
+|                       |      press a Load     |                       |
+|                       |      button to assign |                       |
+|                       |      the same track   |                       |
+|                       |      on both decks.   |                       |
```
Updated

evoixmr commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,830 @@
+|                       |      the knob to      |    | Preview Play     |
+|                       |      cycle between    |      Track            |
+|                       |      the Crates and   |    | SHIFT (right     |
+|                       |      Library. Hold    |      deck)            |
```
Updated

evoixmr commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,830 @@
++-----------------------+-----------------------+-----------------------+
+|                       |    Default Function   |    Shift Function     |
++=======================+=======================+=======================+
+|  1.                   |    | Browse Knob:     |    | Speed Scrolling  |
```
Updated appearance

ronso0 commented:
Re the schematic overview:
correct me if I'm wrong: the Platinum FX is identical to the Pro FX except it has jog wheel displays.
So we can use the Pro FX overview and just change some numbers.

I'll prepare a PR.

ronso0 commented:
Here you go.

I have no idea what's causing the `rstcheck` to fail :shrug:

evoixmr commented:
I just corrected the rstcheck errors.

ronso0 commented:
Re Tap button:
I see you have some logic in place to guess which deck the user wants to address.
Wouldn't it simplify things a lot, i.e. make it prediactbale/reliable for users, to use the left/right deck's **Shift** for this, like you do for the Browse knob?

BPM reset could be mapped to Tap longpress then (required for default 'file BPM' tap, for tempo reset there's the rate slider).

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,363 @@
+   "22", "FX Toggle On / Off:", "Push up on the toggle switch to latch (lock) the FX in the on position. Push down on the toggle switch to turn the FX on momentarily. When the toggle switch is in the middle position, the FX will be off."
+   "23", "Tap BPM", "Press this a minimum of 4 times to manually enter a new BPM. The software will ignore the track's BPM and follow your manually entered tempo.
+
+   *See Tapping Logic*"
```
hint: IIRC you can link to headers in this document:
* add a ref to the header
* link to it with
   ```
   :ref:`LinkTitle<ref-target>`
   ```

For example see `reloop-terminal-mix-series-options` in https://github.com/mixxxdj/manual/blob/2b3021375e075b3a1df77a4a8aa17705a2fefd77/source/hardware/controllers/reloop_terminal_mix_series.rst?plain=1

evoixmr replied:
Done on all the lines with see modes.

evoixmr commented:
How do I correct this error? Is there anything I can do here?

Sphinx / Link Check (pull_request) Failing after 2m
/home/runner/work/manual/manual/source/chapters/appendix/changelog.rst:549: WARNING: broken link: file:line ()

ronso0 commented:
Don't care about that, links may time out occassionally, and you didn't touch that file :shrug: 
(the link in question is https://github.com/mixxxdj/mixxx/pull/12253 and it works for me, formatting also looks okay)

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -56,6 +56,7 @@ Control Mapping
    "3", "Mic Gain", "Adjusts the level for the microphone input."
    "4", "Cue Mix", "Adjusts the audio output to the headphones, mixing between the cue output and the master mix output."
    "5", "Cue Gain", "Adjusts the volume for headphone cueing."
+   "",  ":hwlabel:`SHIFT` Global Sample Volume Adjust", "Turning Cue Gain with Shift held globally adjusts volume on all samples"
```
Let's avoid redundancies : )
```suggestion
   "",  ":hwlabel:`SHIFT` Global Sample Volume Adjust", "Adjusts volume on all sample decks"
```

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -349,7 +350,7 @@ Tapping
 Logic for the tap button:
 
-1. If only one deck is loaded the BPM is control the loaded deck.
+1. If only one deck is loaded, the BPM control will be assigned to theloaded deck.
```
```suggestion
1. If only one deck is loaded, the BPM control will be assigned to the loaded deck.
```

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -189,8 +189,10 @@ Further to the NORMAL Pad Mode, 2 additional layers of functionality, SHIFT & AL
+
    • (LED On)", "Hot Cue 9-16", "Roll Loop", "Fader Cuts 8", "Sample 9-16"
-   ":hwlabel:`Hold` or :hwlabel:`Double Click`
+   ":hwlabel:`Hold` or :hwlabel:`2X PRESS`
```
Somehow I find the hwlabels confusing for this. I'll propose another version..

evoixmr replied:
I updated all the suggestions.

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,407 @@
+   "25", "Scratch", "Press this button to turn on the scratch feature for the jog wheel."
+   "", ":hwlabel:`SHIFT` Deck Select", "Switch between decks 1 & 3 or 2 & 4 for 4 deck mixing"
+   "26", "Shift", "Allows multiple control commands to be triggered when pressed first along with other buttons."
+   "27", "Sync", "**Tap** this button to automatically match the corresponding Deck's tempo with the opposite Deck's tempo."
+   "", ":hwlabel:`HOLD` Lock Sync", "Hold for at least one second to lock sync. Tap again to release sync lock."
```
```suggestion
   "27", "Sync (tap)", "Automatically match the corresponding Deck's tempo with the opposite Deck's tempo."
   "", "Lock Sync (hold)", "Hold for at least one second to lock sync. Tap again to release sync lock."
```

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,407 @@
+   "27", "Sync", "**Tap** this button to automatically match the corresponding Deck's tempo with the opposite Deck's tempo."
+   "", ":hwlabel:`HOLD` Lock Sync", "Hold for at least one second to lock sync. Tap again to release sync lock."
+   "", ":hwlabel:`SHIFT` Quantize", "Toggle quantize on/off."
+   "28", "Cue", "Sets and recalls the main Cue point in the current track"
+   "", ":hwlabel:`HOLD` Cue Play", "Hold down the Cue button for temporary play of the cue point. The track will play for as long as the button is held down and return to the cue point once it is released"
```
```suggestion
   "28", "Cue (tap)", "Sets and recalls the main Cue point in the current track"
   "", "Cue Play (hold)", "Play from the Cue point for as long as the button is held down and return to the cue point once it is released"
```

ronso0 replied:
actually this is standard cue behaviour (CueButton component uses [`cue-default`](https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-cue_default))

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,407 @@
+
+   See :ref:`Performance Pad Modes <numark-platinum-fx-performance-pad-modes>`"
+   "", ":hwlabel:`SHIFT` Sampler 9-16", "Selects layer 2 of samples - Performance pads then trigger samples 9-16."
+   "", ":hwlabel:`HOLD` or :hwlabel:`2X PRESS`
```
```suggestion
   "", "Hold or double-press
```

ronso0 replied:
etc. if you agree with this proposal

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -0,0 +1,407 @@
+
+   See :ref:`Performance Pad Modes <numark-platinum-fx-performance-pad-modes>`"
+   "", ":hwlabel:`SHIFT` Roll", "Each pad triggers a roll of a different length"
+   "", ":hwlabel:`HOLD` or :hwlabel:`2X PRESS`
```
```suggestion
   "", "Hold or double-press
```

ronso0 commented:
Thanks for the reminder @evoixmr 
It looks like work/review on https://github.com/mixxxdj/mixxx/pull/12872 has stalled, but this PR is mostly ready?
What do you think about effect toggles
> Press one of these buttons to select a software effect.
• HPF (High pass filter)
• LPF (Low pass filter)
• Flanger
• Echo
• Reverb
• Phaser
 
IIUC work on this mapping started while 2.3 was the stable version, and 2.4 now allows to load specific effect presets (ie. items at specific position in the preset list, not by effect uuid or something).
Does it make sense to address this now?

evoixmr commented:
> Thanks for the reminder @evoixmr It looks like work/review on [mixxxdj/mixxx#12872](https://github.com/mixxxdj/mixxx/pull/12872) has stalled, but this PR is mostly ready? What do you think about effect toggles
> 
> > Press one of these buttons to select a software effect.
> > • HPF (High pass filter)
> > • LPF (Low pass filter)
> > • Flanger
> > • Echo
> > • Reverb
> > • Phaser
> 
> IIUC work on this mapping started while 2.3 was the stable version, and 2.4 now allows to load specific effect presets (ie. items at specific position in the preset list, not by effect uuid or something). Does it make sense to address this now?

Yes, the PR is ready. As far as he effects preset, is there a documentation that I can use to add preset?

ronso0 commented:
All effect / effect chain preset controls are documented here
https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#controls

Loading specific Quick Effect presets can be done with [`[QuickEffectRack1_[ChannelI]],loaded_chain_preset`](https://manual.mixxx.org/2.4/en/chapters/appendix/mixxx_controls.html#control-[QuickEffectRack1_[ChannelI]]-loaded_chain_preset).
This control expects an integer argument 0 .. [num_chain_presets - 1], so the user has to arrange the Quick Effect presets in Preferences -> Effects to match the button labels (or any desired custom mapping).

ChrisTallon commented:
Please forgive me, I'm getting there slowly. I have been quite confused by the suggestion to use QuickEffects here in this PR (and weirdly in the mapping file in a comment). In Serato Lite the filter knobs (directly under the EQ knobs) only ever activate a filter* and are nothing to do with the 6 FX preset buttons. This is why I was trying to preset the main effects units to match the buttons on the controller.

A fundamental question here is - is this mapping attempting to best reproduce the behaviour of the controller in Serato Lite, or is this mapping trying to match Mixxx? I don't know the answer and I don't have any connection with the original authors.

Possible ways forward:

1) The Mixxx approach. Have the 6 FX preset buttons change which effect the filter knob uses by using [[QuickEffectRack1_[ChannelI]],loaded_chain_preset]. This significantly diverges away from the original intent of the controller. Also this makes the on/off/hold flappy toggle switches redundant unless the user configures the effects units with mouse+keyboard.

2) The Serato approach. Hard-code in presets for the effects units (not QuickEffects) into the init function of the controller. Accept that when the effects list changes (e.g. on a version change of Mixxx) the preset effects may be wrong for new users.

3) The leave-it-as-it-is approach. Alter the manual to inform users that the effects buttons won't function until specific effects are chosen manually. Field complaints in the forum that FX don't work, as currently happens.

There is no good option but I favour 3. After reading the manual, complaining in the forum or just figuring it out, behaviour similar to Serato can be achieved.

Thoughts? Anyone?

( * There is an option to switch it to a "dub echo" effect but it's well hidden enough that I never found it when I used Serato.. )

evoixmr commented:
@ChrisTallon option 3 worked well for all users of the mapping.

@ronso0 what's your thoughts on this?

ronso0 commented:
> In Serato Lite the filter knobs (directly under the EQ knobs) only ever activate a filter*

There is no dedicated Filter effect slot/bank/unit in Mixxx. Effects are available in QuickeEffect unit (just one slot) and standard effect units (currently 3 slots but expandable).
Though Filter is the default Quick Effect so mapping the Super knob of Quick Effect unit N is the way to go for now.

Regarding the labeled effect toggles:
I forgot that the Platinum FX has no (Quick) effect assign buttons, so my reference to the Traktor way of mapping them makes no sense here, however these toggles could be used to assign Quick Effect presets to both decks at once. Is that what's happening in Serato? Or is the selected effect applied to the main output?

So yes, the quick fix is to leave the effect buttons unmapped for now and improve the situation in followup PRs.
(note that is has been some time since I last looked at this and the mapping PR)

ChrisTallon commented:
> Regarding the labeled effect toggles: I forgot that the Platinum FX has no (Quick) effect assign buttons, so my reference to the Traktor way of mapping them makes no sense here, however these toggles could be used to assign Quick Effect presets to both decks at once. Is that what's happening in Serato? Or is the selected effect applied to the main output?

In Serato the 6 effect buttons configure that single effect in slot number 1 of the main effects unit. Then the on/off/hold flappy switches are used to activate it.

The knobs labelled "Filter" directly under the EQs are equivalent to Mixxx's QuickEffects but hard set to "filter". (Or dub-echo if you're lucky enough to find it) and are never affected by the 6 select buttons.

User qgazq appears to have designed the mapping of the Mixtrack controls to Mixxx effects units, and explains here:

https://github.com/qgazq/mixxx_numark_mixtrack_platinum_fx/blob/main/README.md#effects

So, leaving the code as-is, how about this as an addition to the manual?

https://github.com/ChrisTallon/manual/commit/06ff5010fcc434a47f52c189dfe599c65a407f13

Preview: 

https://www.loggytronic.com/mixxx/manual-preview/hardware/controllers/numark_mixtrack_platinum_fx.html#effects

ronso0 commented:
> In Serato the 6 effect buttons configure that single effect in slot number 1 of the main effects unit. Then the on/off/hold flappy switches are used to activate it.
>
> The knobs labelled "Filter" directly under the EQs are equivalent to Mixxx's QuickEffects but hard set to "filter". (Or dub-echo if you're lucky enough to find it) and are never affected by the 6 select buttons.

Thanks, got it!

To get the Serato behavior
* two effect units need to be assigned to Main output (in script init)
* the user has to load the labeled effects in a specific order, or have a specific order in Preferences -> Effects -> Effect Presets
* script init() would load the presets by index
* script would also create a HPF and LPF effect presets (load Filter in two slots, adjust the LPF/HPF parameters)

Besides the required setup, this makes it hard to use these effect units with decks / a specfic deck only, for example with another controller or just the GUI / keyboard, so I'd vote for not making this the (hard-coded) default.

But this is a perfect case for controller settings (example for MIDI mapping: [xml part](https://github.com/mixxxdj/mixxx/blob/cc92ef1263ecdb80bb5935fc4f99e5581648e437/res/controllers/Denon-MC7000.midi.xml#L96-L106), [js part](https://github.com/mixxxdj/mixxx/blob/cc92ef1263ecdb80bb5935fc4f99e5581648e437/res/controllers/Denon-MC7000-scripts.js#L47-L50)): a checkbox "Use effect toggles for Main output" or something.
Or select one of 6 quick effects for both decks. Benefit: not just a flap toggle but also the Filter knob to control the effect
And maybe another alternative mapping, idk.

evoixmr commented:
@ronso0 can we aim for milestone 2.5.1?

ronso0 commented:
Sure we can aim for that : )

This manual still states that the FX buttons select certain preset which is not implemented in https://github.com/mixxxdj/mixxx/pull/12872 (which has stalled), other than that this PR looks good I think.
Once https://github.com/mixxxdj/mixxx/pull/12872 gets LGTM, too, I'll take a final look here (got other stuff on the table atm, sorry).

evoixmr commented:
@ronso0 I updated effect section of the manual and added:
"Press one of these buttons to select the user assigned software effect."
"FX Buttons will need tp be assigned on the mixxx [Effects](https://manual.mixxx.org/2.5/en/chapters/effects#effects) option on the main interface."

ronso0 commented on source/hardware/controllers/numark_mixtrack_platinum_fx.rst:
```diff
@@ -77,13 +77,15 @@ No.      Control                                         Function
                                                          | • Echo
                                                          | • Reverb
                                                          | • Phaser
+                                                         | FX Buttons will need tp be assigned suing the mixxx `Effects <https://manual.mixxx.org/2.5/en/chapters/effects#effects>`__ option on the main interface.
```
```suggestion
                                                         | FX Buttons will need to be assigned with the mixxx `Effects <https://manual.mixxx.org/2.5/en/chapters/effects#effects>`__ selectors in the graphical interface.
```

evoixmr replied:
Updated/

evoixmr commented:
@ronso0  updated.

Swiftb0y commented:
you'll also have to either rebase on `upstream/2.5` or merge the `upstream/2.5` branch to fix the CI.

evoixmr commented:
@Swiftb0y I tried to push the upstream yo 2.5, my pc is having ssh authentication issue,. Is this doable on the browser?

Swiftb0y commented:
I'm afraid not properly, you'll have to fix your ssh setup...

evoixmr commented:
@Swiftb0y I fixed the ssh issue. The rebase to 2.5 was not successful, too much conflict. Any advice?

ronso0 commented:
@evoixmr it seems you merged mixxx/2.5
Did you use `git rebase` or `merge`?
The rebase command would be `git rebase -i mixxx/2.5`
Make sure to fetch mixxx before that.

Oh, do you use the command line at all?

evoixmr commented:
@ronso0 I used the rebase on terminal,error message:

Auto-merging .tx/config
CONFLICT (content): Merge conflict in .tx/config
Auto-merging source/conf.py
error: could not apply 831ab4cfa... Bump version to 2.5
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
hint: Disable this message with "git config advice.mergeConflict false"
Could not apply 831ab4cfa... Bump version to 2.5

evoixmr commented:
Rebase to 2.5

Swiftb0y commented:
Thank you. Preview looks good on first look, but I don't have time for a proper review right now. What about you @ronso0?

ChrisTallon commented:
@evoixmr Do you want me to try to push that effects paragraph I wrote into evoixmr/manual:2.5 ?

https://github.com/ChrisTallon/manual/commit/06ff5010fcc434a47f52c189dfe599c65a407f13

ronso0 commented:
Hmm I can't push as ususal when doing fixups in other's PRs.
We don't want to just close this PR and start anew, so one way forward would be for @evoixmr :
* rename you branch locally `git checkout 2.4` then `git branch -m 2.4_bak`
(and next time don't work on top of a Mixxx branch please)
* create a new branch `git checkout -b 2.4 mixxx/2.5` --> replace 'mixxx' with the name of your Mixxx remote, might be 'upstream' or something
* add my repo as remote
`git remote add ronso0 git@github.com:ronso0/manual.git`
* fetch `git remote fetch ronso0` (now your git knows about all my branches and commits)
* cherry-pick the rebased commit `git cherry-pick 7534d1029c3453d28d91dc3e3e6c2f32bfb706a2`
* force-push to your repo, implicitly updating this PR

ronso0 commented:
that rebased commit is also here #724

ronso0 commented:
Made this a draft to prevent accidental merge.

LGTM btw :+1:

evoixmr commented:
@ronso0 done

> Hmm I can't push as ususal when doing fixups in other's PRs. We don't want to just close this PR and start anew, so one way forward would be for @evoixmr :
> 
> * rename you branch locally `git checkout 2.4` then `git branch -m 2.4_bak`
>   (and next time don't work on top of a Mixxx branch please)
> * create a new branch `git checkout -b 2.4 mixxx/2.5` --> replace 'mixxx' with the name of your Mixxx remote, might be 'upstream' or something
> * add my repo as remote
>   `git remote add ronso0 git@github.com:ronso0/manual.git`
> * fetch `git remote fetch ronso0` (now your git knows about all my branches and commits)
> * cherry-pick the rebased commit `git cherry-pick 7534d1029c3453d28d91dc3e3e6c2f32bfb706a2`
> * force-push to your repo, implicitly updating this PR

ronso0 commented:
Hmm no, apparently not. We still see
![image](https://github.com/user-attachments/assets/f9f83633-f9bd-4de4-897a-2c8e8f2642cc)
Should now be 1 Commit, 3 Files changed

What does this print?
`git log -1 --oneline`
Should be this, just that *** shows your remote state
`7534d1029 *** add mapping for Numark Mixtrack Platinum FX`

evoixmr commented:
@ronso0  4eab7259 (HEAD -> 2.4, origin/HEAD, origin/2.4) add mapping for Numark Mixtrack Platinum FX

Can you check now?

ronso0 commented:
Ah, now you've pushed!
LGTM, except that we should add a white background to the outline SVG so that it can be read on dark backgrounds, too.
Please replace it with this one
![numark_mixtrack_platinum_fx](https://github.com/user-attachments/assets/b1eaf9cf-fd54-4337-9636-75f89d03fc55)
Than `git add`, `git commit --amend` and force-push again please.

evoixmr commented:
> @evoixmr Do you want me to try to push that effects paragraph I wrote into evoixmr/manual:2.5 ?
> 
> [ChrisTallon@06ff501](https://github.com/ChrisTallon/manual/commit/06ff5010fcc434a47f52c189dfe599c65a407f13)

@ChrisTallon not yet,  Itrying to resolve the rebase issues.

ronso0 commented:
Pleeeeease use the command line and squash all into one commit.
`git fetch` to get all commits you did in the web UI
`git reset d4eab7259cdd5f260bf3fa5643d659856ab381c1`
you should the changes with `git status`, should be the SVG only
`git add --all`
`git commit --amend`
and force-push.

evoixmr commented:
@ronso0  Sorry about that., updated on the command line.

ronso0 commented:
Great, LGTM!
Thanks for your patience!

Next time: pick a descriptive branch name like `numark-fix-typos` or something, this will save us some hazzle.
Thank you!

ronso0 commented:
(just letting CI run once more to be sure)

evoixmr commented:
@ronso0  I tried the descriptive branch name. I could not push it from fixing_svg_error to 2.4. It was asking me to do a PR, the PR I open did not see the changes I made.

ronso0 commented:
Don't chage the branch name while there already is a PR with that branch! It'll just create unnecessary trouble.
That's why I wrote "next time".
And you can't change the base branch while a PR is open. You can only change the target branch

> I could not push it from fixing_svg_error to 2.4

You could do that with `git push [remote] localBranch:remoteBranch`. But that's pointless here (see above).
Simply pick a descriptive branch name for your next feature / bugfix.

ronso0 commented:
Oh, apparently pre-commit is not happy with the optimized SVG.
But we can use the patch file provided by CI (see above: 1 failing check, click it, click on Summary at the top left, scrol to bottom, see patch file)

Next git exercise :wink: 
* download the pre-commit patch, copy the location (/path/to/patch.file)
* apply the patch with `git apply /path/to/patch.file`
* `git add --all`
* `git commit --amend`
* force-push

ronso0 commented:
(I presume you don't have pre-commit installed, otherwise you'd have noticed when commiting the SVG I sent)

evoixmr commented:
> Oh, apparently pre-commit is not happy with the optimized SVG. But we can use the patch file provided by CI (see above: 1 failing check, click it, click on Summary at the top left, scrol to bottom, see patch file)
> 
> Next git exercise 😉
> 
> * download the pre-commit patch, copy the location (/path/to/patch.file)
> * apply the patch with `git apply /path/to/patch.file`
> * `git add --all`
> * `git commit --amend`
> * force-push

@ronso0 Done

ronso0 commented:
Thanks!

No, the link error is unrelated and looks like it was a temp. issues, https://www.gnu.org/philosophy/free-sw.html is reachable.

ronso0 commented:
LGTM now, let's go!
